src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -417,15 +417,35 @@ NodeListInsertCommand(List *workerNodeList)
 		char *hasMetadataString = workerNode->hasMetadata ? "TRUE" : "FALSE";
 		char *isActiveString = workerNode->isActive ? "TRUE" : "FALSE";
 
+		char *nodeRoleString = NULL;
+
+		if (workerNode->nodeRole != InvalidOid)
+		{
+			Datum nodeRoleOidDatum = ObjectIdGetDatum(workerNode->nodeRole);
+			Datum nodeRoleStringDatum = DirectFunctionCall1(enum_out, nodeRoleOidDatum);
+			nodeRoleString = DatumGetCString(nodeRoleStringDatum);
+		}
+		else
+		{
+			/*
+			 * Hack because master_initialize_node_metadata calls this before nodeRole
```

Comment:
Is there a way to avoid this hack? (e.g. not calling `NodeListInsertCommand` if there are no workers with metadata in `master_initialize_node_metadata`). Workers with metadata cannot exist at the time `master_initialize_node_metadata` gets called in the extension script.